types/koa-bunyan-logger/index.d.ts

```diff
@@ -0,0 +1,12 @@
+// Type definitions for koa-bunyan-logger 2.0
+// Project: https://github.com/koajs/bunyan-logger
+// Definitions by: Steven McDowall <https://github.com/sjmcdowall>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+/// <reference types="node" />
+import { Middleware } from 'koa';
+
+import * as Logger from 'bunyan';
+
+export default function KoaBunyanLogger(logger: Logger | undefined): Middleware;
```

Comment:
It's not `export default` https://github.com/koajs/bunyan-logger/blob/master/index.js#L261

Reply:
I'm sorry -- but I'm not a guru on converting from whatever form of module declaration is here to what TS should have.. 

What should : 
```
module.exports = function (loggerInstance) {
```
Get translated to in the index.d.ts file to be correct?  I'll happily make the change and re-submit. 

Thanks

Reply:
I think you are missing the idea of the module -- it's not the BUNYAN module we are writing the definition file for, it's the koa-bunyan-logger.  The key line is this:

https://github.com/koajs/bunyan-logger/blob/3ad9bef4faa1df8304ee6828989f2d22df465d27/index.js#L33

That is the core of the module .. The rest is "nice" but not the key bits .. 

As for a test file -- I honestly have no idea how to write a file for a piece of Koa Middleware and I don't see one Koa Middleware @types file with any tests to use as a sample..

Reply:
Then
```js
function KoaBunyanLogger(...

export = KoaBunyanLogger;
```
For the test file, you can use the code from the document

```js
var koa = require('koa');
var koaBunyanLogger = require('koa-bunyan-logger');

var app = koa();
app.use(koaBunyanLogger());

app.use(function *(next) {
  this.log.info('Got a request from %s for %s', this.request.ip, this.path);
  yield next;
});

app.listen(8000);
```